modules/standard/Memory.chpl

```diff
@@ -19,89 +19,42 @@
-    leaked memory by invoking :proc:`printMemAllocsByType` implicitly
-    when the program terminates normally. If there are no leaks,
-    nothing is printed.
-
-  ``memLeaksLog``: `string`:
-    Enable memory tracking and append a report to the named file when
-    the program terminates normally.
-    The report contains the compilation command used to build the
-    program, the execution command used to run it, the summary
-    allocation statistics produced by :proc:`printMemAllocStats`, and the
-    summary information about leaked memory produced by
-    :proc:`printMemAllocsByType`.
-    
-
-  ``memStats``: `bool`:
-    Enable memory tracking and produce summary memory statistics by
-    invoking :proc:`printMemAllocStats` implicitly at normal program
-    termination.  Note that for a multi-locale run, each top-level
-    locale reports its own memory statistics and these reports may
-    appear in any order in the program output (and possibly even
-    interleaved, though we have gone to some effort to avoid that).
-    These statistics can be put in order by sorting the output lines
-    that begin with the string ``memStats:``.
-
-  ``memMax``: `uint`:
-    If the value is greater than 0 (zero), enable memory tracking
-    and use the value as a limit on the number of bytes of memory
-    that can be allocated per top-level (network-connected) locale.
-    If during execution the amount of allocated memory exceeds this
-    limit on any locale, halt the program with a message saying so.
-
-  The following two config variables do not enable memory tracking;
-  they only modify how it is done.
-
-
-  ``memThreshold``: `uint`:
-    If this is set to a value greater than 0 (zero), only allocation
-    requests larger than this are tracked and/or reported.
-
-  ``memLog``: `string`:
-    Memory reporting is written to this file.  By default it is the
-    ``stdout`` associated with the process (not the Chapel channel
-    with the same name).  Setting this config variable to a file path
-    causes the reporting to be written to that named file instead.
-    In multilocale executions each top-level locale produces output
-    to its own file, with a dot ('.') and the locale ID appended to
-    this path.
+  The :mod:`Memory` module provides submodules that contain operations
+  related to memory usage and memory initialization.
+
+  .. warning::
+
+    In previous releases, the :mod:`Memory` module contained procedures
+    which reported information about memory usage. These procedures are now
+    deprecated - please use the equivalents in the :mod:`~Diagnostics`
+    submodule instead.
+
  */
 module Memory {
 
-pragma "insert line file info" private extern proc chpl_memoryUsed(): uint(64);
+include module Diagnostics;
+
+pragma "insert line file info"
+private extern proc chpl_memoryUsed(): uint(64);
 
 /*
+  .. warning::
+
+    This enum is deprecated - please use
+    :enum:`~Diagnostics.MemUnits`.
```

Comment:
Seeing this reminded me of #14396, which I think it would be good for us to prioritize developing with all the module reviews we are doing lately

Reply:
Separately, I think it would be okay to not elide the "Diagnostics" prefix in these references.  It seems like it will make it more obvious to the user where the new version lives, which can help them know how to adjust their use/import statements without having to scroll to the top of the page or look at the new URL when they follow the link, or whatever.  To do so, I think you can just drop the `~`